code/modules/assembly/igniter.dm

```diff
@@ -36,6 +36,12 @@
 	var/turf/location = get_turf(loc)
 	if(location)
 		location.hotspot_expose(heat, EXPOSED_VOLUME)
+	if(istype(loc, /obj/item/assembly_holder))
```

Comment:
Make these signals instead.

You would send a signal to your loc that an igniter is going off, then the assembly holder would likely register the signal itself to send it to *its* loc. Then you register the signal in fueltank to go boom.